pkg/api/annotations.go

```diff
@@ -69,43 +68,26 @@ func PostAnnotation(c *m.ReqContext, cmd dtos.PostAnnotationsCmd) Response {
 		DashboardId: cmd.DashboardId,
 		PanelId:     cmd.PanelId,
 		Epoch:       cmd.Time,
+		EpochEnd:    cmd.TimeEnd,
 		Text:        cmd.Text,
 		Data:        cmd.Data,
 		Tags:        cmd.Tags,
 	}
 
+	if item.EpochEnd == 0 {
+		item.EpochEnd = item.Epoch
+	} else if item.EpochEnd < item.Epoch {
+		tmp := item.Epoch
+		item.EpochEnd = item.Epoch
+		item.Epoch = tmp
+	}
```

Comment:
Guess it's okay to switch times around. Alternative would be to return an error instead.

Reply:
I think the API should swap -- frontend can then drag left or right and not worry

Reply:
Make sense